src/app/store/coronaLocations.reducer.ts

```diff
@@ -0,0 +1,98 @@
+import * as coronaLocationsActions from './coronaLocations.action';
+import { IMapEntity } from '../mapEntity'
+import { Action, ActionReducer, createReducer, on } from '@ngrx/store';
+import { IEntity } from '../entity';
+import { Button } from 'protractor';
+import { state } from '@angular/animations';
+import { ActionType } from 'angular-cesium';
+import { from, Observable } from 'rxjs';
+import { IMockServer } from '@graphql-tools/mock';
+
+export interface State {
+    mapEntities : IMapEntity[];
+    newMapEntities: IEntity[];
+}
+
+const INITIAL_STATE: State = {
+    mapEntities: [],
+    newMapEntities: [],
+}
+
+const _reducer = createReducer(INITIAL_STATE,
+  on(coronaLocationsActions.LOAD_MAP_ENTITIES, (state, {mapEntities}) =>{
+    return{
+      ...state,
+      mapEntities: mapEntities,    
+    };
+  }),
+  on(coronaLocationsActions.ADD_MAP_ENTITY,(state, {entityToAdd})=>{
+    return{
+      ...state,
+      mapEntities: addItem(state.mapEntities, entityToAdd) as IMapEntity[],
```

Comment:
why don't you just add them to newMapEntities, and then in SET_TEMPORARY_LIST move them to mapEntities?